lint_rules/lint_rules/ifs_coding_standards_2011.py

```diff
@@ -472,55 +478,78 @@ class Fortran90OperatorsRule(GenericRule):  # Coding standards 4.15
         '<': re.compile(r'(?P<f77>\.lt\.)|(?P<f90><(?!=))', re.I),
     }
 
+    _op_map = {
+        '==': '.eq.',
+        '/=': '.ne.',
+        '>=': '.ge.',
+        '<=': '.le.',
+        '>': '.gt.',
+        '<': '.lt.'
+    }
+
+    class ComparisonRetriever(Visitor):
+        """
+        Bespoke expression retriever that extracts 3-tuples containing
+        ``(node, expression root, comparison)`` for all :any:`Comparison` nodes.
+        """
+
+        retriever = ExpressionRetriever(lambda e: isinstance(e, sym.Comparison))
+
+        def visit_Node(self, o, **kwargs):
+            """
+            Generic visitor method that will call the :any:`ExpressionRetriever`
+            only on :class:`pymbolic.primitives.Expression` children, collecting
+            ``(node, expression root, comparison)`` tuples for all matches.
+            """
```

Comment:
[no action] Uhhh, nice, although quite advanced interface usage! Still, the level of documentation makes it at least accessible.